### ISSUE TITLE
Back_projection of 3D keypoints

### ISSUE THREAD

**Issue by iulsko** [open-mmlab/mmpose#1708]
hi, how can i implement back-projection of original keypoints onto the image in top_down_video_demo_with_mmdet.py ?

**Comment by ly015** [comment]
Hi, the top-down video demo with mmdet performs 2D keypoint detection from the video and 3D keypoint is not involved. Could you please further explain your needs?

**Comment by iulsko** [comment]
@ly015 hi, i am so sorry, i meant [body3d_two_stage_video_demo.py](https://github.com/open-mmlab/mmpose/blob/master/demo/body3d_two_stage_video_demo.py) and how can i back-project `keypoints_3d`?

**Comment by ly015** [comment]
Please refer to the `camera_to_pixel` method of the [camera class](https://github.com/open-mmlab/mmpose/blob/master/mmpose/core/camera/single_camera.py#L95).

**Comment by iulsko** [comment]
@ly015 this should be applied after [line 458](https://github.com/open-mmlab/mmpose/blob/2b0cd7a3e9cc6b67a08ba697d6683aba6de9aafa/demo/body3d_two_stage_video_demo.py#L458) right?

**Comment by ly015** [comment]
Camera-to-pixel transformation should be applied to the model output at [L442](https://github.com/open-mmlab/mmpose/blob/2b0cd7a3e9cc6b67a08ba697d6683aba6de9aafa/demo/body3d_two_stage_video_demo.py#L442). There are a few technical details that you may need to pay attention to:

- The predicted 3D keypoints could be reletave coordinates around the root (usually pelvis) joint (depending on the model configs)
- The predicted 3D keypoint coordinates could be normalized (depending on the model configs), and you may need to restored their scale before projecting them to the image space
- The argument `norm_pose_2d` should be `False`
- The projected 2D coordinates are in a local coordinate space defined by the bbox

**Comment by iulsko** [comment]
dear @ly015 thank you so much for your answer! i will consider these tips.
i have a small "to understand" question:

in your implementation of VideoPose3D, is the output of `pose_lift_results` same as original implementation e.g. in [line 350](https://github.com/facebookresearch/VideoPose3D/blob/1afb1ca0f1237776518469876342fc8669d3f6a9/run.py#L350)?

**Comment by ly015** [comment]
Yes, we followed the model details and experiment settings from the original paper and implementation. As for the demo script, we added some tricks (e.g. normalize 2D inputs, transform output coordinate space for better visualization, temporal smoothing, et al) for better results, which may differ from the original implementation.

**Comment by iulsko** [comment]
@ly015, I am sorry, I have another question, it is because I am still confused about 3D-back-to-2D, if i have a relative-to-root coordinate situation, how can I bring it back to the camera 3D frame?

**Comment by ly015** [comment]
In this case, the model needs to further predict the absolute coordinates of the root joint in the camera space, or I am afraid it's impossible to project the 3D keypoints back to the image space and completely align with the original image.
